File: toolset/jvm/modules/standalone-plugins/plugin/src/test/java/org/example/GreetingPluginTest.java
package org.example;


import org.gradle.api.Project;
import org.gradle.api.Task;
import org.gradle.testfixtures.ProjectBuilder;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertNotNull;


public class GreetingPluginTest {

    @Test
    public void greeterPluginAddsGreetingTaskToProject() {
        Project project = ProjectBuilder.builder().build();
        project.getPluginManager().apply(GreetingPlugin.class);

        Task task = project.getTasks().findByName("hello");

        assertNotNull(task, "Task 'hello' should not be null");
    }
}

File: toolset/jvm/build-logic/plugins/plugin-tested-one/src/test/kotlin/com/nophasenokill/GreetingPluginTest.kt
package com.nophasenokill

import org.gradle.testfixtures.ProjectBuilder
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertNull
import org.junit.jupiter.api.Test


class GreetingPluginTest {
    @Test
    fun `plugin registers task`() {
        // Create a test project and apply the plugin
        val project = ProjectBuilder.builder().build()
        project.plugins.apply("com.nophasenokill.greetingPlugin")

        // Verify the result
        assertNotNull(project.tasks.findByName("taskInsideGreetingPlugin"))
        assertNull(project.tasks.findByName("taskInsideGreetingPlugin2"))
    }
}
